Inferences/Induction.cpp

```diff
@@ -690,30 +690,34 @@ void InductionClauseIterator::performIntInduction(Clause* premise, Literal* orig
                    ,0))),
                    Formula::quantify(new BinaryFormula(Connective::IMP,FyInterval,Ly)));
   
-  // Create pairs of Literal* and SLQueryResult for resolving L[y] and Y>=b (or Y<=b or Y>b or Y<b)
-  static ScopedPtr<RobSubstitution> subst(new RobSubstitution());
-  // When producing clauses, 'y' should be unified with 'term'
-  subst->unify(TermList(term), 0, y, 1);
-  ResultSubstitutionSP resultSubst = ResultSubstitution::fromSubstitution(subst.ptr(), 1, 0);
-  List<pair<Literal*, SLQueryResult>>* toResolve = new List<pair<Literal*, SLQueryResult>>(
-      make_pair(Ly->literal(), SLQueryResult(origLit, premise, resultSubst)));
+  auto toResolve = List<pair<Literal*, SLQueryResult>>::empty();
   // Also resolve the hypothesis with comparisons with bound(s) (if the bound(s) are present/not default).
   if (!isDefaultBound) {
     // After resolving L[y], 'y' will be already substituted by 'term'.
     // Therefore, the second (and third) substitution(s) is/are empty.
     static ResultSubstitutionSP identity = ResultSubstitutionSP(new IdentitySubstitution());
-    toResolve = List<pair<Literal*, SLQueryResult>>::cons(make_pair(
-          Literal::complementaryLiteral(bound1.literal),
-          SLQueryResult(bound1.literal, bound1.clause, identity)),
-        toResolve);
+    List<pair<Literal*, SLQueryResult>>::push(make_pair(
```

Comment:
Apparently this doesn't work not because the type cannot be deduced from the second argument but rather due to `List` not begin a typename in itself without the template parameters. But surprisingly (to me), `toResolve->push(..., toResolve)` works. Is that good enough or should we leave it?

Reply:
Hmm, maybe even just `push` (no qualifiers) would work because of Koenig lookup. Just a thought, if not I think the original is most clear.

Reply:
Yes, that would be the cleanest. My compiler doesn't find the function this way unfortunately. I guess this is partly why "static" functions in STL like `make_pair`, `make_unique` and so on are defined outside the template classes...

Reply:
Boo. Never mind, thanks for checking!